tests/s3_tester.c

```diff
@@ -1548,35 +1547,7 @@ int aws_s3_tester_send_meta_request_with_options(
             /* if uploading via filepath, write input_stream out as tmp file on disk, and then upload that */
             if (options->put_options.file_on_disk) {
```

Comment:
We don't have any tests for aws_parallel_input_stream_read() failing...

but I guess that's hard to pull off, since the public aws_s3_meta_request_options don't actually take a aws_parallel_input_stream... so we can't make a fake one that fails the Nth time you call read() (like we have for [input-stream](https://github.com/awslabs/aws-c-io/blob/main/include/aws/testing/stream_tester.h) and [async-stream](https://github.com/awslabs/aws-c-io/blob/main/include/aws/testing/async_stream_tester.h)

huh

we could have aws_s3_meta_request_options take an `struct aws_parallel_input_stream *send_parallel_stream`, but marked like `/* FOR TESTING ONLY */` ??

Reply:
GOOD CALLOUT on the the failure case, it caught the bug as we are preparing parts out of data, and we set the part info in the list based on index. So, that if the later parts finish before the early parts and something went wrong, the part info list will have garbage in it.